lib/remote_server/stash.rb

```diff
@@ -254,8 +254,12 @@ def make_request(method, url, args = [])
           body = response.body
           Rails.logger.info("Stash response: #{response.inspect}")
           Rails.logger.info("Stash response body: #{body.inspect}")
-          unless response.is_a? Net::HTTPSuccess
-            raise "#{response.class} body: #{body}"
+          unless response.is_a?(Net::HTTPSuccess)
+            if response.is_a?(Net::HTTPUnauthorized)
+              raise RemoteServer::AccessDenied.new(url, method, body)
```

Comment:
Probably but they can be added as needed—too much pain to go hunting for them all now. The generic raise error is sufficient.